tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -521,6 +519,11 @@ std::vector<std::string> ExpandGlob(const std::string &glob)
    return ret;
 }
 
+std::unique_ptr<TVirtualIndex> MakeIndexCopy(TVirtualIndex *index)
```

Comment:
What is the advantage of using this wrapper function rather than the underlying virtual function directly?

Reply:
I don't want to make all functions in `Internal::TreeUtils` a friend of `TVirtualIndex` and derived

Reply:
Make sense.  The question becomes why is `TVirtualIndex::MakeIndexCopy` not public?

Reply:
Because currently we (TTreeProcessorMT) is the only user of the function and I didn't want to make it public right away, but if you think it's not harmful I can make it public

Reply:
Humm ... This reminded me that one good implementation would have been to simply overload `TObject::Clone` with the content of the current `MakeIndexCopy`.   This allows to do 'less' (see discussion about setting or not setting `fTree`) which breaking the user expectation (of what a 'copy' constructor should do).